app.R

```diff
@@ -263,6 +280,43 @@ server <- function(input, output, session) {
     # Return a list containing the filename and alt text
     list(src = filename, alt = "Camera Image", width = 300)
   }, deleteFile = FALSE)
+  
+  
+  #heatmap values based on user's input 
+  filtered_hm <- reactive({
+    filter_DT(heatmap_data,input$timeRange,as.POSIXct(format(input$dateRange, "%Y-%m-%d")))
+
+  })
+  
+  #heatmap
+  observeEvent({
+    input$timeRange
+    input$dateRange},
+    {                    # do some work in a block and return a leafletProxy
+      hmdff <- filtered_hm()  # get the filtered data frame
+      lfp <-             # and use this data to create the map
+        leafletProxy("basemap", data = hmdff) %>%
+        clearHeatmap() %>%
+        addHeatmap(
+          lng = ~hmdff$longitude,
+          lat = ~hmdff$latitude,
+          max = 14,
+          radius = 5,
+          blur = 3,
+          intensity = ~hmdff$car_count,
```

Comment:
Either:
1. heatmap only display car count, hence the user control - vehicle type should be updated and fixed to car for consistency when switching to heatmap.
2. heatmap can take vehicle type as a variable, hence, truck, bike, etc.